lib/Elastica/ResultSet.php

```diff
@@ -14,117 +14,45 @@
 class ResultSet implements \Iterator, \Countable, \ArrayAccess
 {
     /**
-     * Class for the static create method to use.
-     *
-     * @var string
-     */
-    protected static $_class = 'Elastica\\ResultSet';
-
-    /**
      * Results.
      *
-     * @var array Results
+     * @var Result[] Results
      */
-    protected $_results = array();
+    private $_results = array();
```

Comment:
The main reason for the change here is that the right place to manipulate the results is in the ResultSetBuilder, not here - private forces people to use the defined extension point.